src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/MasterClusterStateUpdateStatsCollector.java

```diff
@@ -0,0 +1,190 @@
+        MetricsProcessor {
+    public static final int SAMPLING_TIME_INTERVAL =
+            MetricsConfiguration.CONFIG_MAP.get(MasterClusterStateUpdateStatsCollector.class).samplingInterval;
+    private static final int KEYS_PATH_LENGTH = 0;
+    private static final Logger LOG = LogManager.getLogger(MasterClusterStateUpdateStatsCollector.class);
+    public static final String MASTER_CLUSTER_UPDATE_STATS_CLASS_NAME = "org.elasticsearch.cluster.service.MasterClusterUpdateStats";
+    private static final String GET_MASTER_CLUSTER_UPDATE_STATS_METHOD_NAME = "getStats";
+    private static final ObjectMapper mapper;
+    private volatile MasterClusterStateUpdateStats prevMasterClusterStateUpdateStats = new MasterClusterStateUpdateStats();
+    private final StringBuilder value;
+    private final PerformanceAnalyzerController controller;
+    private final ConfigOverridesWrapper configOverridesWrapper;
+
+    static {
+        mapper = new ObjectMapper();
+        mapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
+        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    }
+
+    public MasterClusterStateUpdateStatsCollector(PerformanceAnalyzerController controller,
+                                                  ConfigOverridesWrapper configOverridesWrapper) {
+        super(SAMPLING_TIME_INTERVAL, MasterClusterStateUpdateStatsCollector.class.getSimpleName());
+        value = new StringBuilder();
+        this.controller = controller;
+        this.configOverridesWrapper = configOverridesWrapper;
+    }
+
+    @Override
+    public void collectMetrics(long startTime) {
+        if (!controller.isCollectorEnabled(configOverridesWrapper, getCollectorName())) {
+            return;
+        }
+        try {
+            long mCurrT = System.currentTimeMillis();
+            if (ESResources.INSTANCE.getClusterService() == null
+                    || ESResources.INSTANCE.getClusterService().getMasterService() == null) {
+                return;
+            }
+            MasterClusterStateUpdateStats currentMasterClusterStateUpdateStats = mapper.readValue(
+                    mapper.writeValueAsString(getMasterClusterStateUpdateStats()), MasterClusterStateUpdateStats.class);
+
+            MasterClusterStateUpdateMetrics masterClusterStateUpdateMetrics = new MasterClusterStateUpdateMetrics(
+                    computeLatency(currentMasterClusterStateUpdateStats), computeFailure(currentMasterClusterStateUpdateStats));
+
+            value.setLength(0);
+            value.append(PerformanceAnalyzerMetrics.getJsonCurrentMilliSeconds())
+                    .append(PerformanceAnalyzerMetrics.sMetricNewLineDelimitor);
+            value.append(masterClusterStateUpdateMetrics.serialize());
+            saveMetricValues(value.toString(), startTime);
+
+            prevMasterClusterStateUpdateStats = currentMasterClusterStateUpdateStats;
```

Comment:
Let's move this line to #85, just after we have evaluated `computeLatency` and `computeFailure`.

Reply:
done